routing/routing_integration_tests/bicycle_route_test.cpp

```diff
@@ -241,76 +372,114 @@ UNIT_TEST(Spain_Madrid_ElevationsDetour)
+                                   mercator::FromLatLon(68.0212968, 32.9632167), 323.951 /* expectedRouteMeters */);
+}
 
-  /// @todo Make correct tunnels elevation (linear between start and finish).
+UNIT_TEST(Russia_UseTrunk_AvoidGasStationsDetours)
+{
+  /// @todo(pastk): still OM detours into many petrol stations along the trunk.
   CalculateRouteAndTestRouteLength(GetVehicleComponents(VehicleType::Bicycle),
-                                   mercator::FromLatLon(51.3822, -2.3886), {0.0, 0.0},
-                                   mercator::FromLatLon(51.3574666, -2.31526436), 8583.27);
+                                   mercator::FromLatLon(55.9132468, 39.1453188), {0.0, 0.0},
+                                   mercator::FromLatLon(55.9146268, 39.153307), 613.721 /* expectedRouteMeters */);
+}
 
+UNIT_TEST(Netherlands_IgnoreCycleBarrier_WithoutAccess)
+{
+  // There is a barrier=cycle_barrier in the beginning of the route
+  // and it doesn't affect routing as there is no explicit bicycle=no.
+  // https://github.com/organicmaps/organicmaps/issues/3920
+  /// @todo(pastk): such barrier should have a small penalty in routing
+  /// as it slows down a cyclist.
   CalculateRouteAndTestRouteLength(GetVehicleComponents(VehicleType::Bicycle),
-                                   mercator::FromLatLon(51.3822, -2.3886), {0.0, 0.0},
-                                   mercator::FromLatLon(51.3576973, -2.31416085), 11131.6);
+                                   mercator::FromLatLon(51.9960994, 5.67350176), {0.0, 0.0},
+                                   mercator::FromLatLon(51.9996861, 5.67299133), 428.801);
 }
 
-UNIT_TEST(UK_Canterbury_UseDismount)
+UNIT_TEST(UK_ForbidGates_WithoutAccess)
 {
-  CalculateRouteAndTestRouteLength(GetVehicleComponents(VehicleType::Pedestrian),
-                                   mercator::FromLatLon(51.2794435, 1.05627788), {0.0, 0.0},
-                                   mercator::FromLatLon(51.2818863, 1.05725286), 305);
+  // A barrier=gate without explicit access leads to a long detour.
+  // https://www.openstreetmap.org/node/6993853766
+  /// @todo OSRM/Valhalla/GraphHopper ignore such gates,
+  /// see https://www.openstreetmap.org/directions?engine=fossgis_valhalla_bicycle&route=51.3579329%2C-2.3137701%3B51.3574666%2C-2.3152644
+  CalculateRouteAndTestRouteLength(GetVehicleComponents(VehicleType::Bicycle),
+                                   mercator::FromLatLon(51.3579329, -2.3137701), {0.0, 0.0},
+                                   mercator::FromLatLon(51.3574666, -2.31526436), 1149.28);
+}
 
-  /// @todo We have 231 sec bicycle detour here, while using dismount footway is 228 sec by foot :)
+UNIT_TEST(UK_Canterbury_AvoidDismount)
+{
+  // A shortcut via a footway is 305 meters, ETAs are similar, but cyclists prefer to ride!
+  // Check the London_GreenwichTunnel case for when dismounting is reasonable as the detour is way too long.
```

Comment:
AFAIR, this test is from a user claim in EMail. Like "All cyclists in our town use this particular footway" :)

Reply:
Did this user tell whether the cyclists cycle through this footway or they dismount and push their bikes indeed?

Reply:
I.e. if they (illegally) cycle through then we shouldn't adjust to route them there and it'd be better to treat this as an AvoidDismount case indeed (as per my changes in this PR). 

But if they dismount indeed and prefer it over making a pedalling detour then its a legit UseDismount case and we can look into fulfilling it (e.g. increase dismount speed/weight).

Reply:
For now I've just added a comment about this controversial situation..